Source/Core/loadImageFromTypedArray.js

```diff
@@ -0,0 +1,61 @@
+/*global define*/
+define([
+        '../ThirdParty/when',
+        './defined',
+        './DeveloperError'
+    ], function(
+        when,
+        defined,
+        DeveloperError) {
+    "use strict";
+
+    /**
+     * @private
+     */
+    var loadImageFromTypedArray = function(buffer, byteOffset, length, format) {
+        //>>includeStart('debug', pragmas.debug);
+        if (!defined(buffer)) {
+            throw new DeveloperError('buffer is required.');
+        }
+
+        if (!defined(byteOffset)) {
+            throw new DeveloperError('byteOffset is required.');
+        }
+
+        if (!defined(length)) {
+            throw new DeveloperError('length is required.');
+        }
+
+        if (!defined(format)) {
+            throw new DeveloperError('format is required.');
+        }
+        //>>includeEnd('debug');
+
+        var bytes = new Uint8Array(buffer, byteOffset, length);
+        var blob = new Blob([bytes], {
+            type : format
+        });
+
+        return when(blob, function(blob) {
```

Comment:
There's no need to use `when` and `deferred` here.  You can get rid of this entire block and use the existing `loadImage` function to simplify everything.

```javascript
var blobUrl = window.URL.createObjectURL(blob);
return loadImage(blobUrl).always(function(){
    window.URL.revokeObjectURL(blobUrl);
});
```

You might have to pass `false` as a second parameter to `loadImage` if the above doesn't work.